src/kagglehub/cache.py

```diff
@@ -34,15 +34,15 @@ def get_cached_path(handle: Union[ModelHandle, str], path: Optional[str] = None)
         raise ValueError(msg)
 
 
-def get_cached_archive_path(handle: Union[ModelHandle]) -> str:
+def get_cached_archive_path(handle: ResourceHandle) -> str:
```

Comment:
I am bit confused on why change the typer here. should it not stay as ": ModelHandle" since ModelHandle inherits from ResourceHandle and not vice versa?

like how can this line below ever be true:
 if isinstance(handle, ModelHandle):